code/modules/assembly/signaler.dm

```diff
@@ -13,22 +13,34 @@
 	var/frequency = FREQ_SIGNALER
 	var/delay = 0
 	var/datum/radio_frequency/radio_connection
-	var/suicider = null
+	///Holds the mind that commited suicide.
+	var/datum/mind/suicider = null
+	///Holds a reference string to the mob, decides how much of a gamer you are.
+	var/suicide_mob
 	var/hearing_range = 1
 
 /obj/item/assembly/signaler/suicide_act(mob/living/carbon/user)
 	user.visible_message("<span class='suicide'>[user] eats \the [src]! If it is signaled, [user.p_they()] will die!</span>")
 	playsound(src, 'sound/items/eatfood.ogg', 50, TRUE)
-	user.transferItemToLoc(src, user, TRUE)
-	suicider = user
+	moveToNullspace()
+	suicider = user.mind
+	suicide_mob = REF(user)
 	return MANUAL_SUICIDE_NONLETHAL
 
-/obj/item/assembly/signaler/proc/manual_suicide(mob/living/carbon/user)
-	user.visible_message("<span class='suicide'>[user]'s [src] receives a signal, killing [user.p_them()] instantly!</span>")
+/obj/item/assembly/signaler/proc/manual_suicide(datum/mind/suicidee)
+	var/mob/living/user = suicidee.current
+	if(!istype(user))
+		return
+	if(suicide_mob == REF(user))
+		user.visible_message("<span class='suicide'>[user]'s [src] receives a signal, killing [user.p_them()] instantly!</span>")
+	else
+		user.visible_message("<span class='suicide'>[user]'s [src] receives a signal and dies like a gamer!</span>")

```

Comment:
So this'll kill them in a special manner if they tried to do my trick themselves (after this PR is merged)? I approve.

Reply:
Actually messed up the grammar here. Will fix later.